dask_kubernetes/core.py

```diff
@@ -404,7 +404,7 @@ def _delete_pods(self, to_delete):
     def _cleanup_succeeded_pods(self, pods):
         terminated_pods = [p for p in pods if p.status.phase == 'Succeeded']
         self._delete_pods(terminated_pods)
-        return [p for p in pods if p.status.phase != 'Succeeded']
+        return [p for p in pods if p.status.phase not in {'Succeeded', 'Failed'}]
```

Comment:
I think this should be a list `[]` rather than a dictionary `{}`.

Reply:
This is a set literal, which I think is more idiomatic since we check for inclusion and don't care about order.

Reply:
Ah nice I had not seen that syntax before. That makes sense then!